src/rda/map.ts

```diff
@@ -192,49 +311,87 @@ export class MuRDAMapStore<MapRDA extends MuRDAMap<any, any>> implements MuRDASt
     public inverse(rda:MapRDA, action:MuRDATypes<MapRDA>['action']) {
         const { type, data } = action;
         const result = rda.actionSchema.alloc();
-        if (type === 'set' || type === 'restore') {
+        if (type === 'set') {
             const key = data.id;
-            const store = this.stores[key];
-            if (store) {
+            const ids = this.keyToIds[key];
+            if (ids && ids.valid) {
                 result.type = 'restore';
-                result.data = rda.setActionSchema.alloc();
-                result.data.id = rda.keySchema.assign(result.data.id, key);
-                result.data.store = store.serialize(rda.valueRDA, result.data.value);
+                result.data = rda.restoreActionSchema.alloc();
+                result.data.id = key;
+                result.data.attr = 'version';
```

Comment:
This is too generic.  Why keep a version history around?  You can just set the value of the key to whatever the previous value was.